[e2e] Replaces the check for ActivityTestRule

ened opened the pull request:
## Description

Fixes the issue outlined in flutter/flutter#53660.

## Related issues

Fixes flutter/flutter#53660.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

<!-- Links -->
[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

digiter commented on packages/e2e/example/android/app/src/androidTest/java/com/example/e2e_example/MainActivityWithPermissionTest.java:
```diff
@@ -0,0 +1,20 @@
+import org.junit.runner.RunWith;
+
+@RunWith(FlutterTestRunner.class)
+public class MainActivityWithPermissionTest {
```
Would you mind adding a comment explaining the purpose of this test?

ened commented:
@digiter addressed your feedback, thank you for the review. 

FYI: There are other "@Rule" that the E2E test should support but there needs to be some back channel to get the information on the applied rules (or the name of the test if you will) to the Dart side of the E2E test. This should be work for a future PR.

ened commented:
@digiter PR is merged, could you please publish on pub when you get a chance?

cyanglaz commented:
The PR seemed to broke CI. Since it is not published, we are going to revert it for now. @ened Please take a look at the error message I posted in https://github.com/flutter/plugins/pull/2657

ened commented:
@cyanglaz the error seems to come from video_player_web.. not sure what I'm looking for, yet.
